### PR TITLE
fixed the "get_curr_context()" bug by removing it all together

### PR DESCRIPTION
**Pull request by jcessna**
simplified lazy evaluation and tracking of contexts and context names

### CODE REVIEW

File: disdat/fs.py [hunk 1]
```diff
@@ -1055,7 +1001,7 @@ def _get_all_link_frames(self, outer_hfr, local_fs_frames=False, s3_frames=False
 
             next_hf = hf_frontier.pop()
 
-            for fr in next_hf.get_frames(self.get_curr_context()):
+            for fr in next_hf.get_frames(self.curr_context()):
```
Comment (kyocum) [inline disdat/fs.py]: curr_context is a property -- do we need ()?

File: disdat/fs.py [hunk 2]
```diff
@@ -1109,7 +1055,7 @@ def shallow_hfr_copy(self, hfr, new_uuid=None, tags=None, presentation=None):
         need_to_copy = False
 
         # Move files in LINK frames to new local destination
-        for fr in hfr.get_frames(self.get_curr_context()):
+        for fr in hfr.get_frames(self.curr_context()):
```
Comment (kyocum) [inline disdat/fs.py]: same

File: disdat/fs.py [hunk 3]
```diff
@@ -1626,7 +1566,7 @@ def remote_add(self, remote_context, s3_url, force, ):
             None
         """
 
-        ctxt_obj = self.get_curr_context()
+        ctxt_obj = self.curr_context()
```
Comment (kyocum) [inline disdat/fs.py]: don't need () I think

**Comment by kyocum** [comment]
`dsdt context` is no longer highlighting the current context with the code in this branch.

**Comment by jcessna** [comment]
what's the status of this PR?  i lost track...

**Comment by kyocum** [comment]
I dev installed and then tried to `dsdt ls` my context and it threw this error:
(disdat) starlord-6:disdat kyocum$ dsdt ls
Traceback (most recent call last):
  File "/Users/kyocum/Code/disdat/disdat/fs.py", line 342, in load
    return state_dict[self.JsonConfig.ACTIVE_CONTEXT]
KeyError: 'active_context_name'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/kyocum/.virtual_envs/disdat/bin/dsdt", line 11, in <module>
    load_entry_point('disdat', 'console_scripts', 'dsdt')()
  File "/Users/kyocum/Code/disdat/disdat/dsdt.py", line 132, in main
    args.func(args)
  File "/Users/kyocum/Code/disdat/disdat/fs.py", line 1814, in <lambda>
    ls_p.set_defaults(func=lambda args: _ls(fs, args))
  File "/Users/kyocum/Code/disdat/disdat/fs.py", line 1715, in _ls
    tags=common.parse_args_tags(args.tag)):
  File "/Users/kyocum/Code/disdat/disdat/fs.py", line 613, in ls
    data_context = self.curr_context
  File "/Users/kyocum/Code/disdat/disdat/fs.py", line 314, in curr_context
    active_context_name = self.load()
  File "/Users/kyocum/Code/disdat/disdat/fs.py", line 344, in load
    raise RuntimeError("No current local context, please change context with 'dsdt switch'")
RuntimeError: No current local context, please change context with 'dsdt switch'

**Comment by kyocum** [comment]
Fixed and LGTM

**Pull request merged**